src/zarr/storage/remote.py

```diff
@@ -59,6 +90,14 @@ def __init__(
 
         if not self.fs.async_impl:
             raise TypeError("Filesystem needs to support async operations.")
+        if not self.fs.asynchronous:
+            warnings.warn(
+                f"fs ({fs}) was not created with `asynchronous=True`, this may lead to surprising behavior",
+                stacklevel=2,
+            )
```

Comment:
@martindurant - I'm not sure we actually care about this so I'm open to removing it. Without fully understanding how the internals of fsspec work, I'm curious if you think we should encourage folks to give us filesystems with `asynchronous=True`.

Reply:
- not all FSs are async-capable.
- *if* async stores are to be called on awaitable methods, asynchronous=True is the right thing. For some implementations, if makes less difference than it used to (or None).
- I don't know if there's a use case for having fsspec use a separate thread and loop and calling the FS with blocking methods.

Reply:
> * not all FSs are async-capable.

👍 

> * _if_ async stores are to be called on awaitable methods, asynchronous=True is the right thing. For some implementations, if makes less difference than it used to (or None).

By design, we only use awaitable methods. The assumption is that we will never need to use Fsspec's synchronization. 

> * I don't know if there's a use case for having fsspec use a separate thread and loop and calling the FS with blocking methods.

My operating assumption is that we want synchronization to be handled by the highest part of the stack (typically zarr or even something higher). Since Zarr has its own thread/loop in these cases, I hope we can avoid mixing with Fsspec's.